runtime/src/accounts_db.rs

```diff
@@ -5856,10 +5863,20 @@ impl AccountsDb {
                         min = *item;
                     }
                 }
-                let mut account: (StoredMetaWriteVersion, Option<StoredAccountMeta<'_>>) =
-                    (0, None);
-                std::mem::swap(&mut account, &mut current[min_index]);
-                scanner.found_account(&LoadedAccount::Stored(account.1.unwrap()));
+                let found_account = &mut current[min_index];
+                if scanner.filter(
+                    &found_account
+                        .1
+                        .as_ref()
+                        .map(|stored_account| stored_account.meta.pubkey)
+                        .unwrap_or_default(),
```

Comment:
If the stored account is `None`, then the default for the pubkey will be the [System Program](https://explorer.solana.com/address/11111111111111111111111111111111); is that an issue? Is there special handling for this pubkey so that is never is written to the accounts db?